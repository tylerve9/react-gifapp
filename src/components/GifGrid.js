import React from 'react';
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect(() => {
    //     getGifs( category ).then( setImages );
    // }, [ category ]);

    // <div>
    //             { images.map( {id, title} => ( 
    //                 <p key={ img.id } > { img.title } </p> 
    //             )) }
    //         </div>

    const { data: images, loading } = useFetchGifs(category);


    return (
        <div>
            <h3 className="animate__animated animate__fadeInUpBig "> {category} </h3>
            {loading &&
                <p className=" animate__animated animate__fadeInUpBig">loading...</p>}
            {<div className="card-grid">
                {images.map(img => (
                    <GifGridItem key={img.id} {...img} />
                ))}
            </div>}
        </div>
    );

}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}

export default GifGrid;