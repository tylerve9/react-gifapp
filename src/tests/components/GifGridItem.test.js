import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas sobre <GifGridItem />', () => {

    const title = 'Goku';
    const url = 'http://localhost/Goku.jpg';

    const wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    test('Debe mostrar el componente correctamente <GifGridItem />  ', () => {
            
        expect( wrapper ).toMatchSnapshot();
    });

    // test('Debe mostrar los atributos title y url como obligatorios  ', () => {
        
    //     const wrapper2 = shallow( <GifGridItem url={ url } />);
    //     expect( wrapper2 ).not.toMatchSnapshot();
    // });
        
    test('Debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });
        
    test('Debe tener una imagen con el url y el alt de los props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);

    });
        
test('Debe tener una clase llamada animate__fadeInUpBig ', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect( className.includes('animate__fadeInUpBig') ).toBe( true );
})




});