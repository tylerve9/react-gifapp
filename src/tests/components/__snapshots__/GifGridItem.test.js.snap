// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGridItem /> Debe mostrar el componente correctamente <GifGridItem />   1`] = `
<div
  className="card animate__animated animate__fadeInUpBig "
>
  <img
    alt="Goku"
    src="http://localhost/Goku.jpg"
  />
  <p>
     
    Goku
     
  </p>
</div>
`;
